### PR TITLE
[Feature][Sampling] Add AscendC categorical sampling for MRV2

### PR DESCRIPTION
**Pull request by freyfwt**
### What this PR does / why we need it?

Add an AscendC categorical sampling backend for NPU Model Runner V2 (MRV2). It replaces per-vocabulary Gumbel noise generation with row-local Philox randomness and inverse-CDF selection, retaining the existing Triton implementation as the availability fallback.

- Preserve the `logits_cache` contract in the pinned verified vLLM main revision and v0.28.0: cache raw logits before temperature scaling, in the supplied FP16/BF16/FP32 dtype, with scalar/per-token columns and actual cache strides.
- Support FP16/BF16/FP32 logits, greedy and categorical rows, padding, broadcast logits, and vocabularies up to 1,048,576. Accept INT32/INT64 request mappings natively and validate request indices before narrowing.
- Provide `use_fp64=True` through a 64-bit Philox-derived draw and fixed-point mass accumulation, not IEEE FP64 vector arithmetic. Native execution failures propagate; only unavailable native registration selects the old Triton path.
- Validate invalid device data in eager and ACLGraph execution. A deterministic regression now covers FP32 uniform draws rounded to 1.0 on positive-infinity rows.
- Use vector first-index reduction for greedy rows, removing a measured scalar-scan regression while preserving tie-breaking and partial-tile semantics.

This is the first integration stage of #14130, with explicitly narrower scope: ordinary and draft-token sampling use this wrapper; rejection/bonus resampling remains unchanged, including its existing FP64 limitation. The availability fallback and current raw-logits cache contract also supersede the original RFC proposal. Neither `rejection_sampler_utils.py` nor `pr_test.yaml` is changed; CI changes only add three runtime estimates to the existing one-card test route.

### Does this PR introduce _any_ user-facing change?

Yes. Supported MRV2 sampling uses the native backend when available. No new user configuration is added. Sampling remains request-seed/position based, including draft/target stream separation, but exact token streams can differ from Triton; cross-backend bitwise equivalence is not promised. FP64 rejection resampling is not enabled by this PR.

### How was this patch tested?

Measured revision: `295b3cb3a45675b719f45d0931798249eff6f430`; performance refreshed on 2026-09-10.

- Same-revision A3 correctness evidence: **216 passed per version**, zero failures/errors/skips and successful pytest exits, covering native sampling, wrapper/cache behavior, ACLGraph, INT64 mapping boundaries and heavy-tail distributions:
  - verified-main `b2f685834a6456197e7033966fdef52a23f1abcd`;
  - release `v0.28.0` (`2cf0a6915ce544dc493a0990f2ea38d81601128a`).
- This refresh reruns microbenchmarks, not model inference. Qwen3-0.6B MRV2 BF16/eager generation with `use_fp64_gumbel=True` and Python 3.10/3.11/3.12 mypy previously passed at `9efde94`; those are historical evidence, not fresh runs at the measured revision.

Paired A3 microbenchmarks use the pinned verified-main revision, CANN 9.1.0, PyTorch 2.10.0, torch_npu 2.10.0.post4.dev20260715 and triton-ascend 3.2.2. The unchanged upstream Ascend Triton fallback is the baseline; native availability and source/binary provenance are checked. Timings cover the complete sampling primitive (including Triton's `argmax/gather`), **not model throughput**. Inputs, compilation and warmup are excluded; provider order alternates. ACLGraph uses device events, eager uses synchronized wall time; results are medians.

The 108-case random matrix covers B={1,4,16,64,256,512}, V={4096,32000,65536,128256,151936,262144}, FP16/BF16/FP32, temperature=0.7, seed=17. Graph speedup is **15.03–391.43×** (equal-weight geometric mean 152.73×); eager is **10.50–320.93×** (117.97×). Six BF16 greedy anchors improve **2.14–2.44×** graph / **2.16–3.55×** eager. These are same-stack measurements, not portable speedup guarantees.

Representative FP32 random medians (microseconds):

| B × V | Triton graph | Native graph | Graph speedup | Eager speedup |
| --- | ---: | ---: | ---: | ---: |
| 1 × 32000 | 9636.00 | 42.10 | 228.88× | 98.12× |
| 1 × 151936 | 44807.75 | 125.05 | 358.32× | 257.81× |
| 16 × 32000 | 9639.79 | 82.46 | 116.90× | 75.75× |
| 16 × 151936 | 44817.59 | 175.09 | 255.97× | 186.84× |
| 256 × 32000 | 67375.89 | 479.47 | 140.52× | 120.17× |
| 256 × 151936 | 313614.78 | 1101.05 | 284.83× | 266.00× |

All **177 reported configurations** completed, with no default-precision graph/eager regressions against Triton: 108 random, six greedy, 30 cache/strided-cache/draft/INT64-mapping/no-temperature, nine non-aligned/million-vocabulary shapes, six FP64 configurations and 18 reversed-order seed=29 repeats. The independent repeats give 103.33–283.08× graph / 82.38–273.58× eager speedup. Device process occupancy was monitored throughout. An orchestration guard reported a nonzero exit due to a diagnosed own-process exit race; the final 18-case group was independently rerun, with benchmark and monitor exits of 0. All reported benchmark subprocesses exited 0.

FP64 has no supported same-precision Triton baseline. With BF16 inputs, B={1,16,128}, V={32000,151936}, native FP64 graph medians are **0.73–13.16 ms**, costing **9.61–26.46×** the native default path across these six cases; no FP64-versus-Triton speedup is claimed.

Reproduce the main matrix with `python benchmarks/benchmark_categorical_sampling.py --rounds 5 --repeats 2 --output matrix.csv`; the script also supports cache/draft/greedy scenarios, custom shapes, a reversed second-seed run, and separate FP64 timings. CSV output includes medians and graph quartiles.

Hardware registration includes A2/A3/A5, but runtime and performance evidence here is A3 only. The measured native build matches the validated source tree and focuses on categorical sampling plus the RMSNorm dependency; this is not a clean full-operator build claim. The existing one-card CI route is A2; A5 and full cross-chip coverage remain review/merge gates.

- vLLM main: https://github.com/vllm-project/vllm/commit/b2f685834a6456197e7033966fdef52a23f1abcd

### CODE REVIEW

File: csrc/sampling/categorical_sample/op_kernel/categorical_sample.h [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+        GlobalTensor<CacheT> cache;
+        cache.SetGlobalBuffer(reinterpret_cast<__gm__ CacheT*>(logitsCacheGm_));
+        LocalTensor<CacheT> values;
+        if constexpr (IsSameType<CacheT, float>::value) {
+            values = logitsFloat;
+        } else {
+            values = workBuf_.Get<CacheT>();
+            Cast(values, logitsFloat, RoundMode::CAST_RINT, AlignUpU32(validElements, VECTOR_ALIGNMENT_ELEMENTS));
+        }
+        PipeBarrier<PIPE_ALL>();
```
Comment (chenlongxiao) [inline csrc/sampling/categorical_sample/op_kernel/categorical_sample.h:355]: PIPE_ALL 流水全打断？
